akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbIdMapping.scala

```diff
@@ -19,19 +19,30 @@ private[persistence] trait LeveldbIdMapping extends Actor { this: LeveldbStore 
 
   private val idOffset = 10
   private var idMap: Map[String, Int] = Map.empty
```

Comment:
2 things must be updated together, the idMap and the write to leveldb, see `numericId ` that is calling `writeIdMapping`.

I'm pretty sure leveldb is full of locks and synchronized anyway.

Additionally `numericId` is normally a very quick operation (lookup in hashmap), i.e. low contention, apart from when new persistenceIds are created (and then the contention will be the db anyway).